pkg/operator/operator.go

```diff
@@ -112,6 +112,7 @@ func New(
 	}
 
 	deployInformer.Informer().AddEventHandler(optr.eventHandlerDeployments())
+	daemonsetInformer.Informer().AddEventHandler(optr.eventHandler())
```

Comment:
Should we use `eventHandlerDeployments()` instead so that we only reconcile when the DaemonSet is owned by the operator?

Reply:
The initial rationale behind that was to mitigate mao extra reconciling loops we were seeing and make sure the mao didn't watch itself and so racing with the CVO.
I think It's fine to keep this simple and let it watch any daemonSet on the namespace.

Then in the medium term we can consider to polish the mao logic and if we need more granularity we can pass any `SharedInformerOption` with `WithTweakListOptions` to the `NewSharedInformerFactoryWithOptions`
https://github.com/openshift/machine-api-operator/blob/54fa05b00438e3312c23b08d08749f4f004b763e/cmd/machine-api-operator/controller_context.go#L36